src/Domain/ElementIndex.hpp

```diff
@@ -70,17 +76,18 @@ bool operator==(const ElementIndex<VolumeDim>& a,
 template <size_t VolumeDim>
 bool operator!=(const ElementIndex<VolumeDim>& a,
                 const ElementIndex<VolumeDim>& b) noexcept;
+}  // namespace domain
 
 template <size_t VolumeDim>
-size_t hash_value(const ElementIndex<VolumeDim>& index) noexcept;
+size_t hash_value(const domain::ElementIndex<VolumeDim>& index) noexcept;
```

Comment:
Should be in namespace domain for ADL.  I've probably missed some other cases of this.